### ISSUE TITLE
SELinux File Type disagreement

### ISSUE THREAD

**Issue by TJM** [theforeman/puppet-pulp#250]
I am not sure what is disagreeing with puppet, but I have noticed this set of changes *several* times. I am not manually making any changes wrt file contexts.

```
Notice: /Stage[main]/Pulp::Consumer::Config/File[/etc/pulp/consumer/consumer.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Consumer::Config/File[/etc/pulp/agent/conf.d/bind.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Consumer::Config/File[/etc/pulp/agent/conf.d/linux.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Consumer::Config/File[/etc/pulp/agent/conf.d/rpm.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Info: Class[Pulp::Consumer::Config]: Scheduling refresh of Class[Pulp::Consumer::Service]
Info: Class[Pulp::Consumer::Service]: Scheduling refresh of Service[goferd]
Notice: /Stage[main]/Pulp::Consumer::Service/Service[goferd]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Pulp::Admin::Config/File[/etc/pulp/admin/admin.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Config/File[/var/lib/pulp/packages]/seltype: seltype changed 'var_lib_t' to 'httpd_sys_rw_content_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pulp/server.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pki/pulp/content/pulp-global-repo.ca]/seltype: seltype changed 'cert_t' to 'pulp_cert_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pulp/repo_auth.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pulp/server/plugins.conf.d/yum_importer.json]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pulp/server/plugins.conf.d/iso_importer.json]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Config/File[/etc/pki/pulp/rsa.key]/seltype: seltype changed 'cert_t' to 'pulp_cert_t'
Notice: /Stage[main]/Pulp::Apache/File[/etc/pulp/vhosts80/]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Apache/Apache::Vhost[pulp-https]/File[/usr/share/pulp/wsgi]/seltype: seltype changed 'usr_t' to 'httpd_sys_content_t'
Notice: /Stage[main]/Pulp::Apache/Pulp::Apache_plugin[rpm]/File[/etc/pulp/vhosts80/rpm.conf]/seltype: seltype changed 'etc_t' to 'httpd_sys_content_t'
Info: /Stage[main]/Pulp::Apache/Pulp::Apache_plugin[rpm]/File[/etc/pulp/vhosts80/rpm.conf]: Scheduling refresh of Service[httpd]
Notice: /Stage[main]/Apache::Service/Service[httpd]: Triggered 'refresh' from 1 events
```

**Comment by ekohl** [comment]
I've consistently seen this when running the installer a second time. Do you see this going back and forth in production?

My SELinux knowledge is limited. Maybe @lzap knows why this could happen?

**Comment by lzap** [comment]
This is defined in pulp:

/etc/pulp(/.*)? gen_context(system_u:object_r:httpd_sys_content_t,s0)

So our installer puts it into the correct context, but not sure what corrupts this and setting `etc_t` context which is not correct. The only thing that is possible is that these files gets deployed by Puppet before pulp-selinux is installed which carries these file context settings. But then on the next run it's being corrected.

**Comment by ekohl** [comment]
I think PUP-2169 is the underlying issue and https://github.com/puppetlabs/puppet/pull/6278 is open to solve that.

**Comment by ekohl** [comment]
The fix should be part of Puppet 4.10.9. Going to close this but if the issue persists please reopen.
